tests/lib/rules/no-unused-vars.js

```diff
@@ -143,8 +143,15 @@ ruleTester.run("no-unused-vars", rule, {
         {
             code: "try{}catch(err){}",
             options: [{vars: "all", args: "all"}]
-        }
+        },
 
+        // https://github.com/eslint/eslint/issues/6348
+        {code: "var a = 0, b; b = a = a + 1; foo(b);"},
+        {code: "var a = 0, b; b = a += a + 1; foo(b);"},
+        {code: "var a = 0, b; b = a++; foo(b);"},
+        {code: "function foo(a) { var b = a = a + 1; bar(b) } foo();"},
+        {code: "function foo(a) { var b = a += a + 1; bar(b) } foo();"},
+        {code: "function foo(a) { var b = a++; bar(b) } foo();"}
```

Comment:
Can we have test cases for something like:

```js
var a = 2;
a = a * 5 + 6;
```

Should this pass?

Also,

```js
var a = 2, b = 4;
a = a * 2 + b; // b should count as used here
```

If these tests exist already, my apologies, I only read the diff.